airflow/hooks/base_hook.py

```diff
@@ -90,3 +93,78 @@ def get_hook(cls, conn_id: str) -> "BaseHook":
     def get_conn(self) -> Any:
         """Returns connection for the hook."""
         raise NotImplementedError()
+
```

Comment:
This protocol is only infoirmational. But I found it is nice to describe what functionality the custom Hook can have.